src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,60 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public delegate bool TryParseLine(string line, out string parsedValue);
+
+        public async Task<HttpResponseMessage> TryGetLogContents(
+            string logUri,
+            CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
+            {
+                using (var request = new HttpRequestMessage(HttpMethod.Get, logUri))
+                {
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+                    try
+                    {
+                        var response = await _httpClient.SendAsync(request, cancellationToken);
+
+                        if (response.StatusCode == HttpStatusCode.TooManyRequests)
+                        {
+                            if (response.Headers.TryGetValues("Retry-After", out var values) &&
+                                int.TryParse(values.First(), out var seconds))
+                            {
+                                int randomDelay;
+                                lock (_randomNumberGenerator)
+                                {
+                                    randomDelay = _randomNumberGenerator.Next(_retryDelayMin, _retryDelayMax);
+                                }
+                                await Task.Delay(TimeSpan.FromSeconds(seconds).Add(TimeSpan.FromMilliseconds(randomDelay)), cancellationToken);
+                                continue;
+                            }
+                        }
+
+                        response.EnsureSuccessStatusCode();
+                        return response;
+                    }
+                    catch (OperationCanceledException ex) when (ex.CancellationToken == cancellationToken)
+                    {
+                        //Exit if the task got cancelled
+                        throw;
+                    }
+                    catch (Exception)
```

Comment:
```suggestion
                    catch
```

Reply:
We should probably log the exception with a warning that we are swallowing, in case it's interesting.

Reply:
But, again, this code shouldn't exist, use the existing class.

Reply:
using the existing classes now